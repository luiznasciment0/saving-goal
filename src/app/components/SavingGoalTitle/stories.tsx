import { Story, Meta } from '@storybook/react/types-6-0'
import SavingGoalTitle from '.'

export default {
  title: 'SavingGoalTitle',
  component: SavingGoalTitle
} as Meta

export const Default: Story = () => <SavingGoalTitle title="Buy a house" />
