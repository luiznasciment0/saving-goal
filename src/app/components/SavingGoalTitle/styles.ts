import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  place-items: center;
`
export const HomeIcon = styled.img`
  margin-right: 1.6rem;
`
