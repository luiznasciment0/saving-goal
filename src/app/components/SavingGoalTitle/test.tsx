import { render, screen } from '@testing-library/react'

import SavingGoalTitle from '.'

describe('<Typography />', () => {
  it('should render the SavingGoalTitle and its value', () => {
    const { container } = render(<SavingGoalTitle title="Teste" />)

    expect(screen.getByRole('img')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
