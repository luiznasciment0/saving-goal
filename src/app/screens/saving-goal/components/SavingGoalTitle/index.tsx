import HeadingSmallXS from 'app/components/HeadingSmallXS'
import ParagraphXS from 'app/components/ParagraphXS'
import { Wrapper, HomeIcon } from './styles'
import HouseIcon from 'app/assets/images/house-icon.svg'

interface Props {
  title: string
}

const SavingGoalTitle = ({ title }: Props) => {
  return (
    <Wrapper>
      <HomeIcon src={HouseIcon} alt="House Icon" />
      <div>
        <HeadingSmallXS>{title}</HeadingSmallXS>
        <ParagraphXS>Saving goal</ParagraphXS>
      </div>
    </Wrapper>
  )
}

export default SavingGoalTitle
