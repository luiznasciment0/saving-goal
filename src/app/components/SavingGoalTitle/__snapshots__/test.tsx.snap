// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should render the SavingGoalTitle and its value 1`] = `
.c2 {
  font-size: 2rem;
  line-height: 2.4rem;
  font-family: 'Rubik';
  color: #1e2a32;
  font-weight: 500;
}

.c3 {
  font-size: 1.4rem;
  line-height: 2.1rem;
  font-family: 'Work Sans';
  color: #708797;
  font-weight: 400;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  place-items: center;
}

.c1 {
  margin-right: 1.6rem;
}

<div
  class="c0"
>
  <img
    alt="House Icon"
    class="c1"
    src="[object Object]"
  />
  <div>
    <h2
      class="c2"
      color="primaryBlack"
    >
      Teste
    </h2>
    <h3
      class="c3"
      color="primaryGray"
    >
      Saving goal
    </h3>
  </div>
</div>
`
